### PR TITLE
Make e2e tests more configurable

### LINKED ISSUE #437: Make e2e tests more configurable
### Feature request

Currently, the e2e test uses various **hard-coded** parameters for the scenarios around testing with kind clusters only. We do have some environment variables for fetching tokens and ssl certs. Following the same path, we should review other parameters that would make it easier to run tests with different kinds of setups.

Especially the `serverName` and `serverAddress` variables. 

Refer here:

https://github.com/tektoncd/results/blob/fdc47542cc0ce52fadaa4389a01ff0c2b75fe6e5/test/e2e/e2e_test.go#L59-L70

### Use case

- In a case where results is deployed on a remote server
- Results is deployed in a different namespace

### PR DESCRIPTION
**Pull request by avinal**
# Changes

- Add more environment variables to e2e tests to set serverName, serverAddress and certFileName
- Update e2e tests docs with env descriptions
- Fixes #437

/kind misc

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [Tested your changes locally](https://github.com/tektoncd/results/blob/main/docs/DEVELOPMENT.md) (if this is a code change)
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user-facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contain the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/README.md [hunk 1]
```diff
@@ -10,12 +10,26 @@ $ go test --tags=e2e .
...
 
+## E2E Test Environment Variables
+
+The e2e tests use environment variables to modify default values, such as the server name, server address, certificate
+path, etc.The scripts set some of the variables, and you can set other variables to run e2e tests manually.
+
+| Environment variable | Description                                                 | Default                                                       |
+|----------------------|-------------------------------------------------------------|---------------------------------------------------------------| 
+| API_SERVER_ADDR      | The address on which results API server is listening        | https://localhost:8080                                        |
+| API_SERVER_NAME      | Common Name of the server as defined in the SSL certificate | tekton-results-api-service.tekton-pipelines.svc.cluster.local |
```
[the remark this answers is not in the document]
Comment (avinal) [reply test/e2e/README.md:28]: I don't think this will be useful.

**Review by sayan-biswas — APPROVED** [review]

File: test/e2e/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (adambkaplan) [reply test/e2e/README.md:28]: This column just shows the default value. Tekton upstream deploys components into `tekton-pipelines` by default, and I don't see any reason to change that here. Other distributions may choose different namespaces for deployment.

**Review by adambkaplan — APPROVED** [review]
/lgtm

**Pull request merged**